### PR TITLE
Moving from Winston to Log4JS

### PR DESCRIPTION
**Pull request by allenhsieh**
Yanks out Winston because it has lodash as a dependency and uses log4js instead

This pull request makes the following changes:
* Removes Winston
* Adds log4js for logging
* Move all types to DevDeps

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -183,9 +183,7 @@ if (argv) {
         app = new Application(false, silence, artie);
       }
       app.startApplication(argv);
-    } catch (error) {
-      console.error(error.message);
-    }
+    } catch (error) {}
```
Comment (allenhsieh) [inline src/index.ts]: empty catch?

File: src/Services/IqRequestService.ts [hunk 2]
```diff
@@ -34,8 +34,12 @@ export class IqRequestService {
   ) {}
 
   private async init(): Promise<void> {
-    this.internalId = await this.getApplicationInternalId();
-    this.isInitialized = true;
+    try {
+      this.internalId = await this.getApplicationInternalId();
+      this.isInitialized = true;
+    } catch (e) {
```
Comment (allenhsieh) [inline src/Services/IqRequestService.ts]: haha was this to get rid of the unhandled promise errors?

File: src/Application/Logger/Logger.ts [hunk 3]
```diff
@@ -13,90 +13,45 @@
...
 import { existsSync, mkdirSync } from 'fs';
 import { join } from 'path';
 
+export const DEBUG = 'debug';
+export const ERROR = 'error';
+
 const logPath = join(homedir(), '.ossindex');
 
-const logPathFile = process.env.CI ? undefined : join(logPath, '.auditjs.combined.log');
+const logPathFile = join(logPath, '.auditjs.combined.log');
```
Comment (allenhsieh) [inline src/Application/Logger/Logger.ts]: Just confirming we're doing 1 combined log file now

Comment (DarthHater) [reply src/Application/Logger/Logger.ts]: Yep!

File: src/Services/IqRequestService.ts (hunk 2 above)
Comment (DarthHater) [reply src/Services/IqRequestService.ts]: 🤷‍♂

File: src/index.ts (hunk 1 above)
Comment (DarthHater) [reply src/index.ts]: 🤷‍♂

File: src/Services/IqRequestService.ts (hunk 2 above)
Comment (DarthHater) [reply src/Services/IqRequestService.ts]: Yeah, more or less!

File: src/index.ts (hunk 1 above)
Comment (DarthHater) [reply src/index.ts]: Yeah, I dunno why I took out the console.error, etc..., added that back in, great catch!

**Pull request merged**